common/src/main/java/com/wynntils/core/notifications/MessageContainer.java

```diff
@@ -29,4 +41,16 @@ public void editMessage(String newMessage) {
     public void update(MessageContainer other) {
         this.message = other.message;
     }
+
+    public void incrementMessageCount() {
+        this.messageCount++;
+        this.renderedMessage =
+                new TextRenderTask(this.message.getText() + duplicateMessageBuilder(this), this.message.getSetting());
+    }
+
+    private String duplicateMessageBuilder(MessageContainer container) {
```

Comment:
Also, at this point, you can inline this into increamentMessageCount (and then get rid of the if statement, since you always know it is >= 1 after the ++).